internal/controller/dnsrecord_controller_test.go

```diff
@@ -360,7 +335,40 @@ var _ = Describe("DNSRecordReconciler", func() {
 					"ObservedGeneration": Equal(dnsRecord2.Generation),
 				})),
 			)
-			g.Expect(dnsRecord2.Status.WriteCounter).To(BeNumerically(">", int64(0)))
+			g.Expect(dnsRecord2.Status.WriteCounter).To(BeNumerically(">", int64(1)))
+		}, TestTimeoutLong, time.Second).Should(Succeed())
+
+		By("fixing conflict with dnsrecord " + dnsRecord2.Name + " with endpoint dnsName: `foo.example.com` and target: `127.0.0.1`")
+		Eventually(func(g Gomega) {
+			// refresh the second record CR
+			err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsRecord2), dnsRecord2)
+			g.Expect(err).NotTo(HaveOccurred())
+		}, TestTimeoutShort, time.Second).Should(Succeed())
+		dnsRecord2.Spec.Endpoints = getDefaultTestEndpoints()
+		Expect(k8sClient.Update(ctx, dnsRecord2)).To(Succeed())
```

Comment:
Move these inside the eventually. If it fails for some random conflict it will bail out here.

Personal opinion maybe, but i would also much rather see in the spec itself what is being applied to make this work. Just calling `getDefaultTestEndpoints()` really isn't obvious and requires anyone reading this to understand what that is doing in order to know what is being applied. We seem to keep trying to re-factor tests unnecessarily IMO.

Reply:
Moved inside `Eventually()`. 

We are using the same endpoint 4 times and with slight modification 3 more times. It just screams at me "I will break at the very first change to the tests". I made the change with `getDefaultTestEndpoints()` exactly because it took me a while to understand what names/IPs are healthy and which are not. Was intending it as simplification:thinking: but looks like it failed. Would you prefer it being "rolled back"?